pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationManager.java

```diff
@@ -70,6 +70,19 @@ public boolean canProduce(DestinationName destination, String role) throws Excep
         }
     }
 
+    /**
+     * Check if the specified role has permission to access the destination via a proxy
+     *
+     * @param destination
+     *            the fully qualified destination name associated with the destination.
+     * @param role
+     *            the app id used to receive messages from the destination.
+     */
+    public CompletableFuture<Boolean> canProxyAsync(DestinationName destination, String role) {
+        return checkAuthorization(destination, role, AuthAction.proxy);
```

Comment:
instead `AuthAction`, as @merlimat mentioned earlier, should we pass `proxy` as a flag to authorize request?

Reply:
I don't think passing proxy as a flag will work here 

What we want to achieve is restrict the proxy machines (authRole) access to certain topics so that if the proxy is compromised - not all topics can be accessed via the proxy.

Besides creating a separate AuthAction the only other way I see that this will be possible is that per topic along with client role we also add proxy role with AuthAction produce/consume and change the ServerCnx logic to authenticate authRole and originalClientRole both for AuthAction produce/consume each time a producer or consumer is created. 
Advt:- finer granularity in produce and consume
DisAdvt:- More messy code and a bit of an overkill

Anyways - let me know what you think about this,